src/main/java/se/iths/librarysystem/service/BookService.java

```diff
@@ -24,4 +27,12 @@ public BookEntity createBook(BookEntity bookEntity) {
     public Iterable<BookEntity> getBooksByIsbn(String isbn) {
         return bookRepository.findByIsbn(isbn);
     }
+
+    public Optional<BookEntity> findBookById(Long id){return bookRepository.findById(id);}
+
+    public void deleteBook(Long id){
+        Optional<BookEntity> foundItem = Optional.ofNullable(bookRepository.findById(id).orElseThrow(() -> new IdNotFoundException("book", id)));
+        bookRepository.deleteById(foundItem.get().getId());
```

Comment:
```suggestion
        BookEntity foundItem = bookRepository.findById(id).orElseThrow(() -> new IdNotFoundException("book", id));
        bookRepository.deleteById(foundItem.getId());
```

SonarLint is complaining about `foundItem.get().getId()`.